parser.go

```diff
@@ -157,6 +159,26 @@ func (p *parser) parseSRV(rdata []byte) *SRVRecord {
 	return srv
 }
 
+// parseURI parses URI records.
+func (p *parser) parseURI(rdata []byte) *URIRecord {
+	/*
+		                               1  1  1  1  1  1
+		 0  1  2  3  4  5  6  7  8  9  0  1  2  3  4  5
+		+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+		|                   PRIORITY                    |
+		+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+		|                    WEIGHT                     |
+		+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+		|                    TARGET                     |
+		+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+	*/
+	srv := new(URIRecord)
+	srv.Priority = binary.BigEndian.Uint16(rdata[0:2])
+	srv.Weight = binary.BigEndian.Uint16(rdata[2:4])
+	srv.Target = string(rdata[4:])
```

Comment:
RFC7553 states that this is value should be an RFC3986-compliant URI (https://datatracker.ietf.org/doc/html/rfc7553#section-4.4), so I think it would be a good idea to try to parse the result into a URI here. This is a bit annoying because net/url's `Parse` can return an error and the parser's API doesn't currently support this (see https://github.com/babolivier/go-doh-client/issues/10). Feel free to discard the error from `Parse` for now if you feel like adapting the parser's API to return an error is too much of a hassle.